buildSrc/src/main/java/com/google/firebase/gradle/ReleaseGenerator.kt

```diff
@@ -19,53 +19,50 @@ open class ReleaseGenerator : DefaultTask() {
         val pastRelease = project.property("pastRelease").toString()
         val rootDir = project.rootDir
         val availableModules = parseSubProjects(rootDir)
+        val firebaseLibraries = extractLibraries(availableModules)
 
-        val repo = initRepo(rootDir)
+        val repo = Git.open(rootDir)
         val headRef = repo.repository.resolve(Constants.HEAD)
         val branchRef = getObjectRefForBranchName(repo, pastRelease)
 
-        val changedDirs = getChangedLibraries(repo, branchRef, headRef, FIREBASE_LIBRARIES)
-        val changedModules = findChangedModules(changedDirs, availableModules)
-        writeReleaseConfig(rootDir, changedModules, currentRelease)
+        val changedDirs = getChangedLibraries(repo, branchRef, headRef, firebaseLibraries)
+        writeReleaseConfig(rootDir, changedDirs, currentRelease)
     }
 
-    private fun findChangedModules(changedDirs: List<String>, availableModules: Set<String>): List<String> = changedDirs.flatMap { directory ->
-        val moduleName = directory.replace("/", ":")
-        val ktxModuleName = "$moduleName:ktx"
-        listOf(moduleName, ktxModuleName).filter { availableModules.contains(it) }
-    }
+    private fun extractLibraries(availableModules: Set<String>): List<FirebaseLibrary> {
+        val nonKtxModules = availableModules.filter { !it.endsWith("ktx") }.toSet()
+        return nonKtxModules
+                .filter { possibleModule -> !nonKtxModules.any { it.startsWith("$possibleModule:") } }
+                .map { moduleName ->
+                    val ktxModuleName = "$moduleName:ktx"
 
-    private fun parseSubProjects(rootDir: File) = File(rootDir, "subprojects.cfg").readLines().filter { !it.startsWith("#") }.toSet()
+                    val moduleNames = listOf(moduleName, ktxModuleName).filter { availableModules.contains(it) }
+                    val directories = moduleNames.map { it.replace(":", "/") }
 
-    @Throws(IOException::class)
-    private fun initRepo(repoPath: File): Git {
-        if (!repoPath.exists()) {
-            throw RuntimeException("Repo path doesn't exist")
-        }
-        return Git.open(repoPath)
+                    FirebaseLibrary(moduleNames, directories)
+                }
     }
 
+    private fun parseSubProjects(rootDir: File) = File(rootDir, "subprojects.cfg").readLines().filter { !it.startsWith("#") }.filter { it.isNotEmpty() }.toSet()
```

Comment:
Very small nit:

```kotlin
File(project.rootDir, "subprojects.cfg").readLines().filterNot {
      it.startsWith("#") || it.isEmpty()
}.toSet()
```

Might be preferable, depends on taste. But since both filters are checking for the inverse existence of something- it might make more sense.